modules/build/src/main/scala/scala/build/CrossSources.scala

```diff
@@ -462,4 +466,21 @@ object CrossSources {
         ))
     }
   }
+
+  /** When a source file added by a `using file` directive, itself, contains `using file` directives
+    * there should be a warning printed that transitive `using file` directives are not supported.
+    */
+  def warnAboutChainedUsingFileDirectives(
+    sourcesAddedWithDirectives: Seq[PreprocessedSource],
+    logger: Logger
+  ): Unit = for {
+    additionalSource           <- sourcesAddedWithDirectives
+    buildOptions               <- additionalSource.options
+    transitiveAdditionalSource <- buildOptions.internal.extraSourceFiles
+  } do
+    logger.diagnostic(
+      WarningMessages.chainingUsingFileDirective,
```

Comment:
NIT: the warning string could potentially include the name for the `transitiveAdditionalSource` that'd be skipped

Reply:
That's a good suggestion, but I think passing the position parameter to the logger makes it explicit already, see the test assertions.

Reply:
would it list all the transitive dependencies individually like this?
if so, then y're right, that should be sufficient already

Reply:
Will look like this:
```
[warn] .../Bar.scala:1:16
[warn] Chaining the 'using file' directive is not supported, the source won't be included in the build.
[warn] //> using file Foo.scala
[warn]                ^^^^^^^^^
[warn] .../Bar.scala:2:16
[warn] Chaining the 'using file' directive is not supported, the source won't be included in the build.
[warn] //> using file Bar.scala
[warn]                ^^^^^^^^^
```

Reply:
yeah, that's fine then